### DISCUSSION TITLE
[Resolved] 4 Keys as our first POC

### DISCUSSION THREAD

**Discussion by m-linner-ericsson** [cdfoundation/sig-events#42] [General]
During our vocabulary discussion, we raised the question if we should aim for our first POC using the [4 keys project](https://github.com/GoogleCloudPlatform/fourkeys).

To create a POC we would need 3 events (using the ones part of the protocol):
- Change
- Deployment
- Incident

The benefit of using 4 keys project:
- Defined scope
- Defined requirements
- Defined usage
- Existing user of 4 keys can try it and give us feedback

Would creating events supporting 4 keys constitute as a good first candidate for a POC?

**Comment by erkist** [comment]
Personally, based on the type of work I am involved with, I am more interested in releases than deployments, but at least three of the four given metrics (frequency, lead time and change failure rate) seem somewhat valuable for releases as well:

> These four key metrics are:
> 
>    Deployment Frequency
>    Lead Time for Changes
>    Time to Restore Services
>    Change Failure Rate

I will happily support this PoC though.

**Comment by afrittoli** [comment]
`keptn` can create custom events: https://github.com/keptn/spec
We could adapt `Tekton` cloud events to the format.

**Comment by m-linner-ericsson** [comment]
Tried to emphasize we don't want to create additional events but use those proposed.

**Comment by erkist** [comment]
I will classify this discussion as resolved, as we already have a 1st PoC in place. 4 Keys is still highly interesting though, and should not be forgotten.

**Reply by m-linner-ericsson** [reply]
Agree
